api/agent/lb_agent.go

```diff
@@ -26,6 +28,35 @@ type lbAgent struct {
 	shutWg        *common.WaitGroup
 }
 
+type DetachedResponseWriter struct {
+	Headers http.Header
+	Status  int
+	acked   chan struct{}
+}
+
+func (w *DetachedResponseWriter) Header() http.Header {
+	return w.Headers
+}
+
+func (w *DetachedResponseWriter) Write(data []byte) (int, error) {
+	return len(data), nil
+}
+
+func (w *DetachedResponseWriter) WriteHeader(statusCode int) {
+	w.Status = statusCode
+	w.acked <- struct{}{}
```

Comment:
Because the new feature will be somethng not enabled by default and that it is likely that the all-in-one installation will go away, we agreed to have the Detached mode available only (if enabled) with a LB configuration, because of that I think we can leave the ack here. I might thing again to see if there is a better place for the ack channel but for now I leave it here

Reply:
ack seems like it should be in BeforeCall in order to be 'obviously correct' ? regardless of which config (lb/full) we are running in.

Reply:
I don't think we can ack in BeforeCall() in LB. The call has to be forwarded to a runner and runner has to respond with 202.